static/js/components/search/WorkHistoryFilter.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {
+  AnonymousAccessor,
+  TermQuery,
+  SearchkitComponent,
+  RefinementListFilter,
+  NestedBucket,
+  TermsBucket,
+  AggsContainer,
+  CardinalityMetric,
+  FilterBucket,
+} from 'searchkit';
+
+import ModifiedMultiSelect from './ModifiedMultiSelect';
+import type { AvailableProgram } from '../../flow/enrollmentTypes';
+
+export default class WorkHistoryFilter extends SearchkitComponent{
+  props: {
+    currentProgramEnrollment: AvailableProgram,
+  };
+
+  _accessor = new AnonymousAccessor(query => {
+    const { currentProgramEnrollment } = this.props;
+
+    let cardinality = CardinalityMetric("count", 'user_id');
+    let aggsContainer = AggsContainer('company_name_count',{"reverse_nested": {}}, [cardinality]);
+    let termsBucket = TermsBucket(
+      'profile.work_history.company_name',
+      'profile.work_history.company_name',
+      {'size': 20, "order": {"company_name_count": "desc"}},
+      aggsContainer
+    );
+
+    let nestedBucket = NestedBucket('inner', 'profile.work_history', termsBucket);
+    return query.setAggs(FilterBucket(
+      'profile.work_history.company_name11',
+      TermQuery("program.id", currentProgramEnrollment.id),
+      nestedBucket
+    ));
+  });
+
+
+  defineAccessor() {
+    return this._accessor;
+  }
+
+  render() {
+    return (
+      <RefinementListFilter
```

Comment:
It looks like the test is complaining about a lack of an `id` prop. Can you add one? Something like `company_name` should work. You may need to update the accessor above to account for this change